webpack/assets/javascripts/react_app/redux/reducers/toasts/index.js

```diff
@@ -0,0 +1,39 @@
+import {
+  TOASTS_ADD,
+  TOASTS_HIDE,
+  TOASTS_DELETE
+} from '../../consts';
+import Immutable from 'seamless-immutable';
+
+const initialState = Immutable({
+  messages: {},
+  counter: 0,
+  visibilityFilter: 'all'
+});
+
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case TOASTS_ADD: {
+      const id = (state.counter + 1).toString();
```

Comment:
ok..maybe id is not needed? 
Im just trying to think of explicit code we can eliminate in favour of implicit computations